### ISSUE TITLE
EIDF workers

### ISSUE THREAD

**Issue by mjaquiery** [Battery-Intelligence-Lab/galvanalyser#34]
Parts of Galvanalyser functionality should be separated out for deployment _within_ the EIDF. A key part of this is monitoring battery data file uploads and processing those files. 

This will be done on a cluster by creating individual workers who check out and process individual files, creating records in the Galvanalyser database and sending completed records to the EIDF CKAN service for recording and labelling as related to the original file records. 

This process is the **Worker** component in the diagram below.

- [ ] search for unprocessed uploaded files on CKAN
- [ ] 'checkout' a file by registering it in a database
- [x] process a file to extract its data into the generic Galvanalyser format
- [ ] send processed data to CKAN, linked to the raw data as a related record

![restructure](https://user-images.githubusercontent.com/29707911/220093246-0c1d670f-4c41-4d82-aad1-23e7c0cbd21a.png)

**Comment by martinjrobins** [comment]
I don't think we need to have any separate functionality for EIDF in the workers do we? They are still monitoring a directory (just that this directory is in the dat lake). The "checkout" functionality is new, but this would be useful for standard galvanalyser as well I think. And "send processed data to CKAN" is really just inserting into the galvanalyser dbase as normal?

**Comment by mjaquiery** [comment]
Yeah, good idea to make this workflow as generic as possible. Currently harvesters send metadata before they send file data (which creates the dataset), so we could move dataset creation earlier as a checkout step and then populate the metadata and data later.
